src/test/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImplTest.java

```diff
@@ -587,7 +756,16 @@ private void doTestGetCaseByCaseId(CaseType caseType, boolean caseEvents) throws
     CaseRequestDTO requestParams = new CaseRequestDTO(caseEvents);
     CaseDTO results = target.getCaseById(uuid, requestParams);
 
-    CaseDTO expectedCaseResult = createExpectedCaseDTO(caseFromCaseService, caseEvents);
+    List<DeliveryChannel> expectedAllowedDeliveryChannels = null;
+
+    if (caseFromCaseService.getCaseType().equals("SPG") && caseFromCaseService.isHandDelivery()) {
+      expectedAllowedDeliveryChannels = Arrays.asList(DeliveryChannel.SMS);
+    } else {
+      expectedAllowedDeliveryChannels = Arrays.asList(DeliveryChannel.POST, DeliveryChannel.SMS);
+    }
```

Comment:
Again, this can be hardcoded.

Reply:
I'm not sure that it can - this method is called by more than one test. The value of expectedAllowedDeliveryChannels gets set differently depending on what value of caseType is passed in by the calling test.

Reply:
refactored as discussed